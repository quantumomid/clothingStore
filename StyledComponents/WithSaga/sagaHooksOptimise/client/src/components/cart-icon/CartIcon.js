import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { toggleCartHidden } from "../../redux/cart/cartActions";
import { selectCartItemsCount } from "../../redux/cart/cartSelectors";
import { CartIconContainer, ItemCountContainer, ShoppingIcon } from "./cartIconStyles";

const CartIcon = () => {
    const itemCount = useSelector(selectCartItemsCount);
    const dispatch = useDispatch();
    return (
        <CartIconContainer onClick={() => dispatch(toggleCartHidden())}>
            <ShoppingIcon />
            <ItemCountContainer>{itemCount}</ ItemCountContainer>
        </CartIconContainer>
    )
}

export default CartIcon;
