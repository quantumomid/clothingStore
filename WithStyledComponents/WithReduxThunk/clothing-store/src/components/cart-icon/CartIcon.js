import React from "react";
import { connect } from "react-redux";
import { toggleCartHidden } from "../../redux/cart/cartActions";
import { selectCartItemsCount } from "../../redux/cart/cartSelectors";
// import { ReactComponent as ShoppingIcon } from "../../assets/shopping-bag.svg";
// import "./CartIcon.scss";
import { createStructuredSelector } from "reselect";
import { CartIconContainer, ItemCountContainer, ShoppingIcon } from "./cartIconStyles";

const CartIcon = ({ toggleCartHidden, itemCount }) => {
    return (
        <CartIconContainer onClick={toggleCartHidden}>
            <ShoppingIcon />
            <ItemCountContainer>{itemCount}</ ItemCountContainer>
        </CartIconContainer>
    )
}

const mapDispatchToProps = dispatch => ({
    toggleCartHidden: () => dispatch(toggleCartHidden())
});

const mapStateToProps = createStructuredSelector({
    itemCount: selectCartItemsCount
});

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);
