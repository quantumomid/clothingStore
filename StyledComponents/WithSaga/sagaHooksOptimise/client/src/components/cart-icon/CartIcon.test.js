import * as redux from 'react-redux';
import { shallow } from "enzyme";
import CartIcon from './CartIcon';

describe("Cart Icon Component", () => {
    let wrapper;
    let spyOnUseSelector;
    let spyOnUseDispatch;
    let mockDispatch;
  
    beforeEach(() => {
        // Mock useSelector hook
        spyOnUseSelector = jest.spyOn(redux, 'useSelector');
        spyOnUseSelector.mockReturnValue(0);
    
        // Mock useDispatch hook
        spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
        // Mock dispatch function returned from useDispatch
        mockDispatch = jest.fn();
        spyOnUseDispatch.mockReturnValue(mockDispatch);

        wrapper = shallow(<CartIcon />);
    });
  
    afterEach(() => {
        jest.restoreAllMocks();
    });

    it("Cart Icon Component should render", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('Dispatch is called when icon is clicked', () => {
        wrapper.find('CartIconContainer').simulate('click');
        expect(mockDispatch).toHaveBeenCalled();
    });

    it('Should render the itemCount as the text', () => {
        const itemCount = parseInt(wrapper.find('ItemCountContainer').text());
        expect(itemCount).toBe(0);
    });
})