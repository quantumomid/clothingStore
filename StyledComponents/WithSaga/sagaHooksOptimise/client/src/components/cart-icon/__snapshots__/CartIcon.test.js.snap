// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Icon Component Cart Icon Component should render 1`] = `ShallowWrapper {}`;
